src/Dhall/Parser.hs

```diff
@@ -259,24 +259,24 @@ label :: Parser Text
 label = Text.Parser.Token.ident identifierStyle <?> "label"
 
 -- | Parser for a top-level Dhall expression
-expr :: Parser (Expr Src Path)
-expr = exprA
+expr :: Show a => Parser a -> Parser (Expr Src a)
```

Comment:
How about defining:

```haskell
expr :: Parser (Expr Src Path)
expr = exprA embedded
```

... but then re-export `exprA`?  That way end-users can still use `exprA` if they want to customize the parser for embedded values?